KEYCLOAK-10782: Credentials tab on clients can only be displayed with view-realm

rmartinc opened the pull request:
PR for KEYCLOAK-10782. I have requested "query-clients" for those two endpoints, if you think "view-clients" or "manage-clients" should be used instead just let me know (I was not sure what to requests, so I did more or less the same than in KEYCLOAK-9489).

rmartinc committed:
KEYCLOAK-10782: Credentials tab on clients can only be displayed with view-realm

pedroigor commented on services/src/main/java/org/keycloak/services/resources/admin/permissions/RealmPermissionEvaluator.java:
```diff
@@ -62,4 +62,8 @@
+
+    void requireViewPerClientConfigDescription();
+
+    void requireViewClientAuthenticatorProviders();
```
I'm not sure if you need both methods but just a single one ? I would keep `requireViewClientAuthenticatorProviders`.

pedroigor commented on services/src/main/java/org/keycloak/services/resources/admin/permissions/RealmPermissions.java:
```diff
@@ -197,5 +197,18 @@ public void requireViewAuthenticationFlows() {
+    }
+
+    @Override
+    public void requireViewClientAuthenticatorProviders() {
```
Same thing here. I think we should keep `requireViewClientAuthenticatorProviders`.

pedroigor commented on services/src/main/java/org/keycloak/services/resources/admin/permissions/RealmPermissions.java:
```diff
@@ -197,5 +197,18 @@ public void requireViewAuthenticationFlows() {
+
+    @Override
+    public void requireViewClientAuthenticatorProviders() {
+        if (!(canViewRealm() || root.hasOneAdminRole(AdminRoles.QUERY_CLIENTS))) {
```
Don't you also need to include `VIEW_CLIENTS` and `MANAGE_CLIENTS` roles ?

pedroigor commented:
@rmartinc, sorry for the late reply. I think `query-clients` is OK but you also need to consider `VIEW_CLIENTS` and `MANAGE_CLIENTS`.

rmartinc commented:
@pedroigor No problem Pedro, I have added only the  "requireViewClientAuthenticatorProviders" method and the condition now checks the three client perms (I just copied what it was done in the other case). I added some more tests to check that everything works. Check if now it's OK.

pedroigor commented:
@rmartinc looks good now. Thanks for the tests too. I'll run your changes now and see if everything is OK.
